### DISCUSSION TITLE
WebGPU support in WebView2 on Xbox

### DISCUSSION THREAD

**Discussion by AshleyScirra** [MicrosoftEdge/WebView2Feedback#4138] [Q&A]
My testing seems to indicate WebView2 on Xbox One has hardware-accelerated support for WebGL 2, which is good. However WebView2 on Xbox One does not appear to support WebGPU, even though WebView2 on Windows does. Support for WebGPU is potentially important for web games as it allows for better performance and rendering features.

For a minimal repro, just point a WebView2 at this URL: https://webgpu.github.io/webgpu-samples/samples/helloTriangle

It works in WebView2 on Windows, but on Xbox One it reports `TypeError: Cannot read properties of null (reading 'requestDevice')`. Note it does not say that `navigator.gpu` is undefined, so this appears to indicate the WebGPU API is implemented, but no devices are available.

AFAIK Xbox One supports DirectX 12, so in theory WebGPU support is possible as it's designed to work on top of DX12, but perhaps there are some Xbox-specific details that need to be implemented? Perhaps the situation is different on Xbox Series X|S? (I don't have one to test right now.) Is support planned or should I file a bug? Any insight would be appreciated!

**Comment by victorhuangwq** [comment]
@jennifer-jacobs do you have any insights into whether WebGPU is supported? If it's not supported, @AshleyScirra do open a feature request so that we can track it.

**Reply by AshleyScirra** [reply]
I filed https://github.com/MicrosoftEdge/WebView2Feedback/issues/4150 as a feature request.
